core/src/main/java/org/ironjacamar/core/security/DefaultSubjectFactory.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * IronJacamar, a Java EE Connector Architecture implementation
+ * Copyright 2014, Red Hat Inc, and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the Eclipse Public License 1.0 as
+ * published by the Free Software Foundation.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the Eclipse
+ * Public License for more details.
+ *
+ * You should have received a copy of the Eclipse Public License
+ * along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.ironjacamar.core.security;
+
+import org.ironjacamar.core.spi.security.SubjectFactory;
+
+import java.security.Principal;
+
+import javax.resource.spi.security.PasswordCredential;
+import javax.security.auth.Subject;
+
+/**
+ * Implements a basic subject factory
+ */
+public class DefaultSubjectFactory implements SubjectFactory
+{
+   /** The security domain name */
+   private String securityDomain;
+
+   /** The user name */
+   private String userName;
+
+   /** The password */
+   private String password;
+
+   /**
+    * Create a new DefaultSubjectFactory.
+    */
+   public DefaultSubjectFactory()
+   {
+      this(null, null, null);
+   }
+
+   /**
+    * Create a new DefaultSubjectFactory.
+    *
+    * @param securityDomain securityDomain
+    * @param userName userName
+    * @param password password
+    */
+   public DefaultSubjectFactory(String securityDomain, String userName, String password)
+   {
+      this.securityDomain = securityDomain;
+      this.userName = userName;
+      this.password = password;
+   }
+
+   /**
+    * Set the security domain
+    * @param v The value
+    */
+   public void setSecurityDomain(String v)
+   {
+      this.securityDomain = v;
+   }
+
+   /**
+    * Set the user name
+    * @param v The value
+    */
+   public void setUserName(String v)
+   {
+      this.userName = v;
+   }
+
+   /**
+    * Set the password
+    * @param v The value
+    */
+   public void setPassword(String v)
+   {
+      this.password = v;
+   }
+
+   @Override
+   public Subject createSubject()
```

Comment:
This method needs to be removed from the SPI, and hence the implementation moved to createSubject(String)